### PR TITLE
New metric erd_accounts_snapshot_num_nodes

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- new metric is needed in order to provide the max number of nodes for trie sync
  
## Proposed changes
- added new metric `erd_accounts_snapshot_num_nodes`

## Testing procedure
- proper tests on feat branch with all components done

## Pre-requisites

Based on the [Contributing Guidelines](CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: trie/statistics/trieStatisticsCollector.go [hunk 1]
```diff
@@ -55,6 +55,11 @@ func (tsc *trieStatisticsCollector) Print() {
 	)
 }
 
+// GetNumNodes returns the number of nodes
```
Comment (iulianpascalau) [inline trie/statistics/trieStatisticsCollector.go:58]: I know this is a problem not related to your PR but we need to fix this:
L36 can easily panic if I send a nil object. Either nil check or change the definition of the Add function to pass the argument by value

Comment (sstanculeanu) [reply trie/statistics/trieStatisticsCollector.go:58]: pushed, passing by value now

File: state/snapshotStatistics.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ func (ss *snapshotStatistics) AddTrieStats(trieStats *statistics.TrieStatsDTO) {
 	ss.mutex.Lock()
 	defer ss.mutex.Unlock()
 
-	ss.trieStatisticsCollector.Add(trieStats)
+	ss.trieStatisticsCollector.Add(*trieStats)
```
Comment (iulianpascalau) [inline state/snapshotStatistics.go:55]: this can easily panic if I send the trie stats as nil. We need to change here also

Comment (sstanculeanu) [reply state/snapshotStatistics.go]: pushed, kept using pointer but added check if nil

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**